src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -797,7 +798,11 @@ protected RouteRequest buildRequest(MultivaluedMap<String, String> queryParamete
           filterBuilder.addSelect(SelectRequest.of().withTransportModes(tModes).build());
         }
 
-        transit.setFilters(List.of(filterBuilder.build()));
+        if (tModes.isEmpty()) {
```

Comment:
Are you that this is gonna work as intended with code from above?

```
        if (modes == null) {
          tModes = MainAndSubMode.all();
        } else {
          // Create modes
          tModes = modes.getTransitModes().stream().map(MainAndSubMode::new).toList();
        }
```

Reply:
yes, because `mode=null` means that you haven't specified the URL parameter. We interpret that as "WALK,TRANSIT".

Reply:
Nvm, it should be OK